test/accel/IntegrationTestBase.cc

```diff
@@ -0,0 +1,410 @@
+ */
+auto IntegrationTestBase::make_tracking_action() -> UPTrackAction
+{
+    return nullptr;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Create optional stepping action (local, default null).
+ */
+auto IntegrationTestBase::make_stepping_action() -> UPStepAction
+{
+    return nullptr;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Create Celeritas setup options.
+ */
+SetupOptions IntegrationTestBase::make_setup_options()
+{
+    celeritas::SetupOptions opts;
+
+    // NOTE: these numbers are appropriate for CPU execution and can be set
+    // through the UI using `/celer/`
+    opts.max_num_tracks = 1024;
+    opts.initializer_capacity = 1024 * 128;
+
+    // Use a uniform (zero) magnetic field
+    opts.make_along_step = celeritas::UniformAlongStepFactory();
+
+    // Save diagnostic file to a unique name
+    opts.output_file = this->make_unique_filename(".out.json");
+    return opts;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Create an optional thread-local sensitive detector.
+ */
+auto IntegrationTestBase::make_sens_det(std::string const&) -> UPSensDet
+{
+    return nullptr;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Create physics list: default is EM only using make_physics_input.
+ */
+auto LarSphereIntegrationMixin::make_physics_input() const -> PhysicsInput
+{
+    PhysicsInput result = Base::make_physics_input();
+    result.em_bins_per_decade = 5;
+    return result;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Create a 10 MeV electron primary.
+ */
+auto LarSphereIntegrationMixin::make_primary_input() const -> PrimaryInput
+{
+    using MevEnergy = Quantity<units::Mev, double>;
+
+    PrimaryInput result;
+    result.pdg = {pdg::electron()};
+    result.energy = inp::MonoenergeticDistribution{MevEnergy{10}};
+    result.shape = inp::PointDistribution{from_cm({99, 0.1, 0})};
+    result.angle = inp::IsotropicDistribution{};
+    result.num_events = 4;
+    result.primaries_per_event = 10;
+    return result;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Create THREAD-LOCAL sensitive detectors.
+ */
+auto LarSphereIntegrationMixin::make_sens_det(std::string const& sd_name)
+    -> UPSensDet
+{
+    EXPECT_EQ("detector", sd_name);
+
+    return std::make_unique<SimpleSensitiveDetector>(sd_name);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Create physics list: default is EM only using make_physics_input.
+ */
+auto TestEm3IntegrationMixin::make_physics_input() const -> PhysicsInput
+{
+    using MevEnergy = Quantity<units::Mev, double>;
+
+    PhysicsInput result = Base::make_physics_input();
+    result.em_bins_per_decade = 14;
+    result.min_energy = MevEnergy{0.1};
```

Comment:
Can you remind me what the `min_energy` option does in practice?

Reply:
Ahhh okay. I was just trying with this command to reduce the simulation time (but inadvertently caused more problems). Let me just revert the min_energy change here and we can add a check in the physics importer.